File: python/leap/leap.py

def is_leap_year(year):
    is_leap = False

    if year % 4 == 0 and year % 100 != 0:
        is_leap = True
    elif year % 100 == 0 and year % 400 == 0:
        is_leap = True

    return is_leap

File: go/leap/leap.go
package leap

// IsLeapYear validate whether the given year is a leap year.
func IsLeapYear(year int) bool {
    if year % 4 == 0 {
        if year % 100 != 0 {
            return true
        }
        if year % 400 == 0 {
            return true
        }
    }
    return false
}
